File: .vendor/dislin/go/src/surfcp.go
package main

import "dislin"
import "math"

func myfunc (x float64, y float64, iopt int) float64 {
  var xv float64

  if iopt == 1 {
     xv = math.Cos (x) * (3 + math.Cos (y))
  } else if iopt == 2 {
     xv = math.Sin (x) * (3 + math.Cos (y))
  } else {
     xv = math.Sin (y)
  }
  return xv
}
 
func main () {

  ctit1 := "Surface Plot of the Parametric Function"
  ctit2 := "[COS(t)*(3+COS(u)), SIN(t)*(3+COS(u)), SIN(u)]"
  pi  := 3.1415927

  dislin.Metafl ("cons")
  dislin.Setpag ("da4p")
  dislin.Disini ()
  dislin.Pagera ()
  dislin.Complx ()

  dislin.Titlin (ctit1, 2)
  dislin.Titlin (ctit2, 4)

  dislin.Axspos (200, 2400)
  dislin.Axslen (1800, 1800)

  dislin.Name   ("X-axis", "X")
  dislin.Name   ("Y-axis", "Y")
  dislin.Name   ("Z-axis", "Z")
  dislin.Intax  ()

  dislin.Vkytit (-300)
  dislin.Zscale (-1.0,1.0)
  dislin.Surmsh ("on")

  dislin.Graf3d (-4.0,4.0,-4.0,1.0,-4.0,4.0,-4.0,1.0,-3.0,3.0,-3.0,1.0)
  dislin.Height (40)
  dislin.Title  ()

  step := 2.0 * pi / 30.0
  dislin.Surfcp (myfunc, 0.0, 2.0 * pi, step, 0.0, 2.0 * pi, step)
  dislin.Disfin ()
}

File: .vendor/dislin/ruby/surfcp.rb
#! /usr/bin/env ruby
require 'dislin'

def myfunc(x, y, iopt)
  if iopt == 1
     xv = Math.cos(x)*(3+Math.cos(y))
  elsif iopt == 2
     xv = Math.sin(x)*(3+Math.cos(y))
  else
     xv = Math.sin(y)
  end

  return xv
end

ctit1 = "Surface Plot of the Parametric Function"
ctit2 = "[COS(t)*(3+COS(u)), SIN(t)*(3+COS(u)), SIN(u)]"

pi  = 3.1415927

Dislin.metafl('cons')
Dislin.setpag('da4p')
Dislin.disini()
Dislin.pagera()
Dislin.complx()

Dislin.titlin(ctit1, 2)
Dislin.titlin(ctit2, 4)

Dislin.axspos(200, 2400)
Dislin.axslen(1800, 1800)

Dislin.name('X-axis', 'X')
Dislin.name('Y-axis', 'Y')
Dislin.name('Z-axis', 'Z')
Dislin.intax()

Dislin.vkytit(-300)
Dislin.zscale(-1.0,1.0)
Dislin.surmsh('on')

Dislin.graf3d(-4.0,4.0,-4.0,1.0,-4.0,4.0,-4.0,1.0,-3.0,3.0,-3.0,1.0)
Dislin.height(40)
Dislin.title()

step = 2 * pi / 30.0

Dislin.surfcp('myfunc', 0.0, 2*pi, step, 0.0, 2*pi, step)
Dislin.disfin()

